src/middlewared/middlewared/plugins/failover.py

```diff
@@ -936,26 +935,10 @@ def upgrade_pending(self):
             self.middleware.call_sync('keyvalue.set', 'HA_UPGRADE', False)
             return False
 
-        local_bootenv = self.middleware.call_sync(
-            'bootenv.query', [('active', 'rin', 'N')])
+        if not self.middleware.call_sync('failover.call_remote', 'bootenv.query', [[('active', '=', 'NR')]]):
+            raise CallError('Remote system will reboot to a boot environment than the one that is currently active')
```

Comment:
I got with our documentation team to see how best to convey this scenario and this is the verbiage they suggest.
```suggestion
            raise CallError('Remote controller must reboot to activate pending boot environment')
```